specification/batchai/resource-manager/Microsoft.BatchAI/stable/2018-05-01/BatchAI.json

```diff
@@ -0,0 +1,3924 @@
+      "enum": [
+        "requeue",
+        "terminate",
+        "waitforjobcompletion",
+        "unknown"
+      ],
+      "x-ms-enum": {
+        "name": "DeallocationOption",
+        "modelAsString": false
+      }
+    },
+    "ScaleSettings": {
+      "properties": {
+        "manual": {
+          "$ref": "#/definitions/ManualScaleSettings",
+          "title": "The scale for the cluster by manual settings"
+        },
+        "autoScale": {
+          "$ref": "#/definitions/AutoScaleSettings",
+          "title": "The scale for the cluster by autoscale settings"
+        }
+      },
+      "description": "At least one of manual or autoScale settings must be specified. Only one of manual or autoScale settings can be specified. If autoScale settings are specified, the system automatically scales the cluster up and down (within the supplied limits) based on the pending jobs on the cluster."
+    },
+    "AutoScaleSettings": {
+      "properties": {
+        "minimumNodeCount": {
+          "type": "integer",
+          "format": "int32",
+          "title": "Specifies the minimum number of compute nodes the cluster can have."
+        },
+        "maximumNodeCount": {
+          "type": "integer",
+          "format": "int32",
+          "title": "Specifies the maximum number of compute nodes the cluster can have."
+        },
+        "initialNodeCount": {
+          "type": "integer",
+          "format": "int32",
+          "title": "Specifies the number of compute nodes to allocate on cluster creation. Note that this value is used only during cluster creation.",
+          "default": 0
+        }
+      },
+      "required": [
+        "minimumNodeCount",
+        "maximumNodeCount"
+      ],
+      "description": "The system automatically scales the cluster up and down (within minimumNodeCount and maximumNodeCount) based on the pending and running jobs on the cluster."
+    },
+    "ManualScaleSettings": {
+      "properties": {
+        "targetNodeCount": {
+          "type": "integer",
+          "format": "int32",
+          "title": "The desired number of compute nodes in the Cluster.",
+          "default": 0,
+          "description": "Default is 0. If autoScaleSettings are not specified, then the Cluster starts with this target."
+        },
+        "nodeDeallocationOption": {
+          "title": "Determines what to do with the job(s) running on compute node if the Cluster size is decreasing.",
+          "description": "The default value is requeue.",
+          "default": "requeue",
+          "$ref": "#/definitions/DeallocationOption"
+        }
+      },
+      "required": [
+        "targetNodeCount"
+      ],
+      "description": "Manual scale settings for the cluster."
+    },
+    "VirtualMachineConfiguration": {
+      "properties": {
+        "imageReference": {
+          "title": "Reference to OS image.",
+          "$ref": "#/definitions/ImageReference"
+        }
+      },
+      "description": "Setting
```

Comment:
By default,. C# has the exact same structure for `CloudError` which is set as the `Body` property in `CloudException`. If this was the intention here, it would be easier to simply not specify the error model for the operations .

Reply:
The reason we did not use CloudError is because we will be adding more properties to Job and Cluster errors. for e.g. category -- userError, serverError. Also we will be returning collection of errors.